tests/metrics/test_stratification.py

```diff
@@ -1,316 +1,175 @@
 
-    months = set(sg.month)
-    assert len(months) == 1
-    assert list(months)[0] == str(stratification_time.month)
-
-
-def test_group(stratifier_and_sim, default_levels, include_and_exclude):
-    rs, sim = stratifier_and_sim
-    sim.step()
-
-    include, exclude = include_and_exclude
-    expected_levels = (set(default_levels) | set(include)) - set(exclude)
-
-    pop = sim.get_population()
-    groups = list(rs.group(pop.index, include, exclude))
-
-    _check_groups(rs, expected_levels, groups, pop)
-
-
-def test_group_empty_index(stratifier_and_sim, default_levels, include_and_exclude):
-    rs, sim = stratifier_and_sim
-    sim.step()
-
-    include, exclude = include_and_exclude
-    expected_levels = (set(default_levels) | set(include)) - set(exclude)
-
-    pop = sim.get_population()
-    pop = pop[pop.age > 200]
-    groups = list(rs.group(pop.index, include, exclude))
 
-    _check_groups(rs, expected_levels, groups, pop)
+def test_results_stratifier_map_age_groups():
+    """Test that ages of the population are mapped to intervals as expected."""
+    pop = pd.DataFrame(FAKE_POP_AGE_DICT)
+    rs = ResultsStratifier()
+    rs.age_bins = pd.DataFrame(AGE_BINS_EXPECTED_DICT)
+    mapped_pop = rs.map_age_groups(pop)
+    assert mapped_pop.to_list() == FAKE_POP_EXPECTED_AGE_BINS_LIST
 
 
-def _check_groups(stratifier, expected_levels, groups, pop):
-    expected_num_groups = 1
-    for level in expected_levels:
-        if level not in ["month", "year"]:
-            expected_num_groups *= len(stratifier.stratification_levels[level].categories)
+def test_results_stratifier_map_year():
+    """Test that datetimes are mapped to the correct year."""
+    pop = pd.DataFrame(FAKE_POP_EVENT_TIME)
+    rs = ResultsStratifier()
+    the_year = rs.map_year(pop)
+    assert (the_year == 2045).all()
 
-    assert len(groups) == expected_num_groups
 
-    if not expected_levels:
-        label, group_mask = groups[0]
-        assert label == ""
-        assert pop[group_mask].equals(pop)
-    else:
-        for label, group_mask in groups:
-            for level in expected_levels:
-                assert level in label
-            assert pop[group_mask].index.isin(pop.index).all()
+def test_results_stratifier_get_age_bins(mocker):
+    """Test that get_age_bins produces expected age_bins DataFrame."""
+    builder = mocker.Mock()
+    builder.data.load = fake_data_load_population_age_bins
+    builder.configuration.population.age_start = 0.0
+    builder.configuration.population.exit_age = 5.0
+    builder.configuration.time.start.year = 2022
+    builder.configuration.time.end.year = 2025
 
+    rs = ResultsStratifier()
+    age_bins = rs.get_age_bins(builder)
 
-# todo test age_stratification_mapper
-# todo test year_stratification_mapper
-# todo test year_current_categories_getter
-# todo test StratificationLevel init
+    assert age_bins.to_dict() == AGE_BINS_EXPECTED_DICT
```

Comment:
Same question as above except with DataFrames and dicts.

Reply:
I found it easier to define the expected values as dicts or lists.

Reply:
I've been able to change this.